code/modules/projectiles/energy_bolt.dm

```diff
@@ -240,55 +222,48 @@ toxic - poisons
 	icon_state = "detain-projectile"
 	power = 20
 	cost = 50
-	dissipation_rate = 6
+	dissipation_rate = 5
+	dissipation_delay = 3
 	color_red = 255
 	color_green = 165
 	color_blue = 0
-	var/distance = 6		//Distance needs to be explicitly set before usage. in shoot// currently only used for the Lawgiver
+	max_range = 7 //slight range boost
 	var/hit = 0				//This hit var and the on_hit on_end nonsense was to make it so that if it hits a guy, the explosion starts on them and not one tile before, but if it hits a wall, it explodes on the floor tile in front of it
 
-	//die/detonate when you go the required distance
-	tick(var/obj/projectile/O)
-		if (distance <= 0)
-			O.die()
-		distance--
 
-	on_hit(atom/O)
+	on_hit(atom/O, angle, var/obj/projectile/P)
```

Comment:
Instead of adding a `var/obj/projectile/P` argument to `on_hit()` and `detonate()`, you could set the projectile datum's `master` var to the projectile object upon initialization. Maybe don't set this in the parent `initialize_projectile()` because it doesn't seem to be necessary in most projectiles' cases, but you could create an override, call the parent (`. = ..()` to set the return value), and then set it.

Reply:
Nevermind, this issue of not using `master` is embed in the parent projectile datum and its procs and would require refactoring that's outside the scope of this PR. If I really care so much about this, I'll do it myself (lol no). Ignore my comment.